docs: forwardref usage for standalone components (#50355)
angular/angular: aio/content/guide/dependency-injection-in-action.md

This isn't usually a problem, especially if you adhere to the recommended *one class per file* rule.
But sometimes circular references are unavoidable.

> You're in a problematic situation when class 'A' refers to class 'B' and 'B' refers to 'A'.

Reviewer: nit: we make it sound a bit more positive :)
```suggestion
For example, when class 'A' refers to class 'B' and 'B' refers to 'A'.
```
